### PR TITLE
look for existing editor input

### LINKED ISSUE #232444: dirty notebook opens if you close the IW before connection completes
1. set `"jupyter.interactiveWindow.notebookRepl": true`
2. run "Create Interactive Window"
3. press ctrl+W before kernel connection completes (1-2 seconds)

🐛 untitled notebook opens in new tab

### PR DESCRIPTION
**Pull request by amunger**
fix https://github.com/microsoft/vscode/issues/232444

### CODE REVIEW

File: src/vs/workbench/contrib/replNotebook/browser/repl.contribution.ts [hunk 1]
```diff
@@ -3,52 +3,54 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { SyncDescriptor } from '../../../../platform/instantiation/common/descriptors.js';
```
Comment (amunger) [inline src/vs/workbench/contrib/replNotebook/browser/repl.contribution.ts:6]: sorted imports

File: src/vs/workbench/contrib/replNotebook/browser/repl.contribution.ts [hunk 2]
```diff
@@ -129,17 +133,34 @@ export class ReplDocumentContribution extends Disposable implements IWorkbenchCo
 					const scratchpad = this.configurationService.getValue<boolean>(NotebookSetting.InteractiveWindowPromptToSave) !== true;
 					const ref = await this.notebookEditorModelResolverService.resolve({ untitledResource: resource }, 'jupyter-notebook', { scratchpad, viewType: 'repl' });
 
+					const notebookUri = ref.object.notebook.uri;
```
Comment (rebornix) [inline src/vs/workbench/contrib/replNotebook/browser/repl.contribution.ts:136]: from what I read, we are not caching the editor inputs anywhere else, e.g. text editor, custom editor, notebook. I wonder if this is any misunderstanding of how it should be cached.

**Comment by amunger** [comment]
This PR still has some issues to investigate, but wanted to confirm that it is needed first.

File: src/vs/workbench/contrib/replNotebook/browser/repl.contribution.ts (hunk 2 above)
Comment (amunger) [reply src/vs/workbench/contrib/replNotebook/browser/repl.contribution.ts:144]: text editors do cache inputs, but they might currently be the only one

**Review by rebornix — APPROVED** [review]

**Pull request merged**